ir/instr.cpp

```diff
@@ -4023,6 +4041,35 @@ StateValue X86IntrinBinOp::toSMT(State &s) const {
 
     return rty->aggregateVals(vals);
   }
+  case mmx_punpckhbw:
+  case mmx_punpckhwd:
+  case mmx_punpckhdq:
+  case mmx_punpcklbw:
+  case mmx_punpcklwd:
+  case mmx_punpckldq:
+  {
+    vector<StateValue> vals;
+    unsigned laneCount, startVal, endVal;
+    switch (op) {
+    case mmx_punpckhbw: laneCount = 8; startVal = laneCount / 2; endVal = laneCount; break;
+    case mmx_punpckhwd: laneCount = 4; startVal = laneCount / 2; endVal = laneCount; break;
+    case mmx_punpckhdq: laneCount = 2; startVal = laneCount / 2; endVal = laneCount; break;
+    case mmx_punpcklbw: laneCount = 8; startVal = 0; endVal = laneCount / 2; break;
+    case mmx_punpcklwd: laneCount = 4; startVal = 0; endVal = laneCount / 2; break;
+    case mmx_punpckldq: laneCount = 2; startVal = 0; endVal = laneCount / 2; break;
+    default: UNREACHABLE();
+    }
+    //Starts at first lane of high half of both vectors
+    for (unsigned i = startVal; i != endVal; ++i) {
+      auto ai = aty->extract(av, i);
+      auto bi = bty->extract(bv, i);
+      
+      vals.emplace_back(move(ai.value), move(ai.non_poison));
```

Comment:
just use `vals.emplace_back(move(ai));` to eliminate the redundant unpacking/packing.

Reply:
Beautiful. That makes it a lot cleaner.